### DISCUSSION TITLE
How to install MOOSE on ARM64 HPC with Fujitsu A64FX cpu?

### DISCUSSION THREAD

**Discussion by zhang-rite** [idaholab/moose#17855] [Q&A Getting Started] [answered]
Hi MOOSE developer,

Is there a method for installing MOOSE on ARM64 HPC with Fujitsu A64FX CPU?
For the ARM64 architecture, the options for compiling may need to change but I don't know how to do it. I have tried following the installation guide `hpc_install_moose` but not successful. 

The HPC has a Fujitsu compiler "Fujitsu Development Studio". 
GCC 11 may also work but I haven't tested it yet. 
Does anyone have the experience? or give any suggestion? Thank you.

**Comment by milljm** [comment, marked as answer]
We have no instructions for supporting ARM, but several folks over the years have successfully built MOOSE on such an architecture (Raspberry Pi, Beaglebone, etc). Almost always it will be about building everything from scratch (GCC, MPICH, their dependencies, etc). And when you do hit a failure along the way, you take those errors and look them up on the internet for a possible solution. Your google-fu will be tested!

It will certainly be an adventure, with no quick path forward...

**Reply by makeclean** [reply]
I've built it on the cavium thunder x2, and it was fine, just stick to the gcc toolchain and I think you'll be fine.

**Comment by GiudGiud** [comment]
Shikhar installed MOOSE on an ARM 64 architecture and explained the steps in this thread
https://github.com/idaholab/moose/discussions/17416
